src/h3/mod.rs

```diff
@@ -1371,8 +1532,513 @@ mod tests {
+            .unwrap();
+
+        s.pipe.advance(&mut buf).ok();
+
+        s.server
+            .send_response(&mut s.pipe.server, stream2, &resp, true)
+            .unwrap();
+
+        s.pipe.advance(&mut buf).ok();
+
+        s.server
+            .send_response(&mut s.pipe.server, stream3, &resp, true)
+            .unwrap();
+
+        s.pipe.advance(&mut buf).ok();
+
+        s.poll_assert(Role::Client, Ok((stream1, Event::Headers(resp.to_vec()))));
+        s.poll_assert(Role::Client, Ok((stream1, Event::Finished)));
+
+        s.poll_assert(Role::Client, Ok((stream2, Event::Headers(resp.to_vec()))));
+        s.poll_assert(Role::Client, Ok((stream2, Event::Finished)));
+
+        s.poll_assert(Role::Client, Ok((stream3, Event::Headers(resp.to_vec()))));
+        s.poll_assert(Role::Client, Ok((stream3, Event::Finished)));
+        s.poll_assert(Role::Client, Err(Error::Done));
+    }
+
+    #[test]
+    fn send_body_invalid_stream() {
+        let mut buf = [0; 65535];
+
+        let mut s = Session::default().unwrap();
+        s.handshake(&mut buf).unwrap();
+
+        // client-side streams
+        assert_eq!(
+            s.client.send_body(&mut s.pipe.client, 0, b"Uh oh", true),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.control_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.local_qpack_streams.encoder_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.local_qpack_streams.decoder_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.peer_control_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.peer_qpack_streams.encoder_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        assert_eq!(
+            s.client.send_body(
+                &mut s.pipe.client,
+                s.client.peer_qpack_streams.decoder_stream_id.unwrap(),
+                b"Uh oh",
+                true
+            ),
+            Err(Error::WrongStream)
+        );
+
+        // server-side streams
+        assert_eq!(
```

Comment:
Do we really need to duplicate the same tests for both sides? I'd expect that the same code handles both cases anyway.

Reply:
Change the test to focus on server.

If client-side tests are different/needed, we can add a different test later.